src/components/Login.js

```diff
@@ -14,10 +18,15 @@ export class Login extends Component {
   handleLogin = e => {
     e.preventDefault()
     const { email, password } = this.state
-    const artist = this.state.artists
-    let mailConfirm = artist.filter(art => art.email === email &&
-      art.password === password)
-    alert(mailConfirm.length > 0
+    const artists = this.state.artists 
+    const clients = this.state.clients
```

Comment:
Acá puedes simplificar un poco haciendo un destructuring `const { artists, clientes } = this.state`